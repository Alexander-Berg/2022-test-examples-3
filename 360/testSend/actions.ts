import { makeCreateAction } from '@/utils/actions';

import {
    ITestSendTask,
    IFanValidationError,
    IFanNotReadyError,
    IFanSendLimitError,
    IFanWrongDomainError,
    IFanWrongLoginError,
} from '@/../common/types';

import { FEATURE_NAME } from './constants';

export interface ITestSendStartPayload {
    recipients: string[];
    campaign_slug: string;
    user_template_variables: Record<string, string>;
}

const createAction = makeCreateAction(FEATURE_NAME);

export const testSend = createAction<string>('testSend');
export const testSendStart = createAction<ITestSendStartPayload>('testSendStart');
export const testSendAbort = createAction<void>('testSendAbort');
export const testSendSuccess = createAction<ITestSendTask>('testSendSuccess');
export const testSendFailure = createAction<
IFanValidationError | IFanNotReadyError | IFanWrongDomainError | IFanSendLimitError | IFanWrongLoginError | undefined
>('testSendFailure');

export {
    notifySuccess,
    notifyFailure,
    notifyGeneralFailure,
    showNotification,
} from '@/features/notifications/actions';
