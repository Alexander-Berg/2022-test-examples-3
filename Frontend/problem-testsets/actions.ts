import { createAsyncAction, createStandardAction, ActionType } from 'typesafe-actions';

import {
    IAddTestsetFailure,
    IAddTestsetRequest,
    IAddTestsetResponse,
    IFetchFailure,
    IFetchRequest,
    IFetchResponse,
    IFetchTestsetFailure,
    IFetchTestsetRequest,
    IFetchTestsetResponse,
    IRemoveFailure,
    IRemoveRequest,
    IRemoveResponse,
    IUpdateRequest,
} from 'client/store/problem-testsets/types';

export const fetchTestsets = createAsyncAction(
    'problems/testsets/GET_REQUEST',
    'problems/testsets/GET_SUCCESS',
    'problems/testsets/GET_FAILURE',
)<IFetchRequest, IFetchResponse, IFetchFailure>();

export const refreshTestsets = createStandardAction(
    'problems/testsets/REFRESH_REQUEST',
)<IFetchRequest>();

export const updateTestset = createAsyncAction(
    'problems/testsets/UPDATE_REQUEST',
    'problems/testsets/UPDATE_SUCCESS',
    'problems/testsets/UPDATE_FAILURE',
)<IUpdateRequest, IFetchTestsetResponse, IFetchTestsetFailure>();

export const fetchTestset = createAsyncAction(
    'problems/testsets/GET_TESTS_REQUEST',
    'problems/testsets/GET_TESTS_SUCCESS',
    'problems/testsets/GET_TESTS_FAILURE',
)<IFetchTestsetRequest, IFetchTestsetResponse, IFetchTestsetFailure>();

export const removeTestset = createAsyncAction(
    'problems/testsets/DELETE_REQUEST',
    'problems/testsets/DELETE_SUCCESS',
    'problems/testsets/DELETE_FAILURE',
)<IRemoveRequest, IRemoveResponse, IRemoveFailure>();

export const addTestset = createAsyncAction(
    'problems/testsets/ADD_REQUEST',
    'problems/testsets/ADD_SUCCESS',
    'problems/testsets/ADD_FAILURE',
)<IAddTestsetRequest, IAddTestsetResponse, IAddTestsetFailure>();

export const addSample = createAsyncAction(
    'problems/testsets/ADD_EXAMPLE_REQUEST',
    'problems/testsets/ADD_EXAMPLE_SUCCESS',
    'problems/testsets/ADD_EXAMPLE_FAILURE',
)<IAddTestsetRequest, IAddTestsetResponse, IAddTestsetFailure>();

const actions = {
    addSample,
    addTestset,
    fetchTestsets,
    refreshTestsets,
    updateTestset,
    fetchTestset,
    removeTestset,
};

export type ProblemTestsetsActions = ActionType<typeof actions>;

export default actions;
