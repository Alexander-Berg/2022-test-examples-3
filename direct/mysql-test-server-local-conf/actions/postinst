#!/bin/bash

AUTO_INCREMENT_OFFSET=0

if hostname | grep -q ppctest-devtest-mysql; then
    AUTO_INCREMENT_OFFSET=1
elif hostname | grep -q ppctest-dev7-mysql; then
    AUTO_INCREMENT_OFFSET=2
elif hostname | grep -q ppctest-ts1-mysql; then
    AUTO_INCREMENT_OFFSET=3
elif hostname | grep -q ppctest-ts2-mysql; then
    AUTO_INCREMENT_OFFSET=4
elif hostname | grep -q ppctest-load-mysql; then
    AUTO_INCREMENT_OFFSET=5
fi

if [ $AUTO_INCREMENT_OFFSET != 0 ]; then
    echo "Writing auto_increment settings for ppcdict, auto_increment_offset = $AUTO_INCREMENT_OFFSET"
    echo -e "[mysqld]\nauto-increment-increment = 5\nauto-increment-offset = $AUTO_INCREMENT_OFFSET" > \
        /etc/mysql/ppcdict.conf.d/auto-increment.cnf
fi

