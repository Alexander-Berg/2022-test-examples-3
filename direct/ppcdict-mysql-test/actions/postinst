#!/bin/bash -e
update-rc.d mysql.ppcdict defaults >/dev/null
