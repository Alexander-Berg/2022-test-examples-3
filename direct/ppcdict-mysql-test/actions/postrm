#!/bin/bash -e
update-rc.d -f mysql.ppcdict remove >/dev/null
