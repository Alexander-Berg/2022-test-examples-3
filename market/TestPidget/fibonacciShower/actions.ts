export const FIBONACCI_LOAD_NEXT = 'TestPidget_fibonacciLoadNext';
export const FIBONACCI_LOAD_NEXT_SUCCESS = 'TestPidget_fibonacciLoadNextSuccess';

export const actions = {
    loadNext: () => ({
        type: FIBONACCI_LOAD_NEXT,
    }),

    loadNextSuccess: (nextFib: number) => ({
        type: FIBONACCI_LOAD_NEXT_SUCCESS,
        payload: nextFib,
    }),
};
