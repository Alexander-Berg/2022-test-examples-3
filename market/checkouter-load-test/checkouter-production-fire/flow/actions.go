package flow

import (
	"a.yandex-team.ru/library/go/slices"
	"a.yandex-team.ru/market/capi/checkouter-load-test/checkouter-production-fire/abstractions"
	"a.yandex-team.ru/market/capi/checkouter-load-test/checkouter-production-fire/clients"
	"a.yandex-team.ru/market/capi/checkouter-load-test/checkouter-production-fire/configs"
	"a.yandex-team.ru/market/capi/checkouter-load-test/checkouter-production-fire/configs/stocktype"
	"a.yandex-team.ru/market/capi/checkouter-load-test/checkouter-production-fire/model"
	"context"
	"fmt"
	"go.uber.org/zap"
	"net/url"
	"strconv"
)

func getStocksFromReport(client clients.ShootingClients, shootContext abstractions.ShootContext, options configs.Options) (chan clients.GetStockResult, context.CancelFunc) {
	stocks, stopStocksRequests := client.GetReport().GetStocks(shootContext, options)
	filtered := make(chan clients.GetStockResult, cap(stocks))
	go func() {
		defer close(filtered)
		for stock := range stocks {
			if stock.Stock == nil {
				continue
			}
			if stock.Err != nil {
				stockLogger(*stock.Stock, stock.Offer, true).Warn(stock.Err.Error())
				continue
			}
			if !options.IsAllowedWarehouseID(stock.Stock.WarehouseID, stocktype.Fulfillment, stocktype.FulfillmentCashback) {
				stockLogger(*stock.Stock, stock.Offer, true).Warn("not suitable warehouseId")
				continue
			}
			amounts, err := client.GetStockStorage().
				GetAvailableAmounts(shootContext, []clients.SSItem{stock.Stock.SSItem})
			if err != nil {
				panic(err)
			}
			if len(amounts) == 0 {
				stockLogger(*stock.Stock, stock.Offer, true).Warn("empty amounts")
				continue
			}
			if amounts[0].Amount < options.ChooseOfferThreshold {
				stockLogger(*stock.Stock, stock.Offer, true).With(zap.Int("quantity", amounts[0].Amount)).
					Warn("quantity is less than threshold")
				continue
			}
			stock.Stock.Quantity = amounts[0].Amount
			filtered <- stock
		}
	}()

	return filtered, stopStocksRequests
}

func getStocksFromYQL(client clients.ShootingClients, shootContext abstractions.ShootContext, options configs.Options) (chan clients.GetStockResult, context.CancelFunc) {
	resp, err := client.GetYQLStocksClient().GetStocks(shootContext, options)
	if err != nil {
		panic(fmt.Errorf("cannot execute getStocks: '%v'", err))
	}
	return clients.StocksArrayToGetStockResult(resp.Stocks), func() {}
}

func GetFeedID(client clients.ShootingClients, shootContext abstractions.ShootContext, options configs.Options,
	supplierID int, isDropship bool, stock clients.Stock) (int, error) {
	reportResult, err := client.GetReport().GetShopInfo(shootContext, supplierID)
	if err != nil {
		return -1, err
	}
	return ExtractFeedIDFromReport(options, supplierID, isDropship, reportResult.ReportShops, stock), nil
}

func GetOfferInfo(client clients.ShootingClients, shootContext abstractions.ShootContext, options configs.Options, feshofferID string, stock clients.Stock) (*clients.OfferInfo, error) {
	resp, err := client.GetReport().GetOfferInfo(shootContext, options, feshofferID)

	if err != nil {
		return nil, fmt.Errorf("cannot execute offer_info: '%v'", err)
	}

	return ConvertStockOfferFromReport(options, resp.ReportSearch, stock), nil
}

func CoinsForCart(client clients.ShootingClients, ammo configs.CustomAmmo, shootContext abstractions.ShootContext, options configs.Options, offers []clients.OfferInfo, cartsCount int, offersCount int) (*clients.СoinsForCart, error) {
	var items []clients.CartItem
	for cartID := 0; cartID < cartsCount; cartID++ {
		items = append(items, getCartItemsFromOffers(offersCount, offers, cartID)...)
	}
	orderItemsRequest := clients.OrderItemsRequest{Items: items}
	resp, err := client.GetLoyaltyClient().FetchBonusesForCart(shootContext, ammo.UID, options.RegionID, orderItemsRequest)
	if err != nil {
		return nil, fmt.Errorf("cannot execute /coins/cart/WHITE/v2: '%w'", err)
	}
	return resp, nil
}

func Cart(client clients.ShootingClients, ammo configs.CustomAmmo, shootContext abstractions.ShootContext, options configs.Options, offers []clients.OfferInfo, cartsCount int, offersCount int, coinsForCart *clients.СoinsForCart) (*clients.CartRootResponse, error) {

	var internalCarts []clients.CartRequest

	var coins []int

	if coinsForCart != nil {
		coinsResponse := *coinsForCart
		for _, coin := range coinsResponse.ApplicableCoins {
			coins = append(coins, coin.ID)
		}
	}

	for cartID := 0; cartID < cartsCount; cartID++ {

		items := getCartItemsFromOffers(offersCount, offers, cartID)

		var shopID int
		if ammo.IsDropship() {
			shopID = offers[0].ShopID
		} else {
			shopID = options.ShopID
		}
		paymentMethod := "CASH_ON_DELIVERY"
		paymentType := "POSTPAID"
		internalCarts = append(internalCarts,
			clients.CartRequest{
				ShopID: shopID,
				Items:  items,
				CartDelivery: clients.CartDelivery{
					RegionID: options.RegionID,
					Address:  options.Addresses[cartID],
				},
				PaymentMethod: &paymentMethod,
				PaymentType:   &paymentType,
				Buyer: clients.Buyer{
					LastName:   "последнееимя",
					FirstName:  "первоеимя",
					MiddleName: "среднееимя",
					Phone:      "+77777777777",
					Email:      options.Email,
					IP:         "8.8.8.8",
					DontCall:   true,
				},
			})
	}

	if ammo.UsePromo && ammo.Promocode == nil {
		panic(fmt.Errorf("missing promocode parameter"))
	}

	req := &clients.CartRootRequest{
		BuyerRegionID: options.RegionID,
		BuyerCurrency: "RUR",
		Cart:          internalCarts,
		Promocode:     ammo.Promocode,
		CoinIdsToUse:  &coins,
	}

	resp, err := client.GetCheckouter().Cart(shootContext, req, options, ammo.UID, ammo.UseFlash)
	if err != nil {
		return nil, fmt.Errorf("cannot execute cart: '%w'", err)
	}

	return resp, nil
}

func getCartItemsFromOffers(offersCount int, offers []clients.OfferInfo, cartID int) []clients.CartItem {
	var items []clients.CartItem
	for offerID := 0; offerID < offersCount; offerID++ {
		offer := offers[offersCount*cartID+offerID]
		items = append(items,
			clients.CartItem{
				FeedID:     offer.FeedID,
				OfferID:    offer.OfferID,
				ShowInfo:   offer.ShowInfo,
				Count:      1,
				BuyerPrice: offer.Price,
			})
	}
	return items
}

func Checkout(client clients.ShootingClients, ammo configs.CustomAmmo, shootContext abstractions.ShootContext, options configs.Options, deliveryPayment []*clients.CheckoutParams, cartRootResponse *clients.CartRootResponse, coinsForCart *clients.СoinsForCart) *clients.CheckoutResponse {

	var orderRequests []clients.OrderRequest

	for cartID := 0; cartID < len(cartRootResponse.Cart); cartID++ {
		cartResponse := cartRootResponse.Cart[cartID]
		var coinIdsToUse *[]int
		if ammo.UseCoins {
			var logMsg string
			coinIdsToUse, logMsg = GetAllowedCoins(cartRootResponse, coinsForCart)
			shootContext.GetLogger().Warn(logMsg)
		} else {
			coinIdsToUse = nil
		}
		for i := range cartResponse.CartItems {
			cart := &cartResponse.CartItems[i]
			if cart.BuyerPriceNominal != nil {
				cart.BuyerPrice = *cart.BuyerPriceNominal
				cart.BuyerPriceNominal = nil
			}
		}
		orderRequests = append(orderRequests,
			clients.OrderRequest{
				ShopID:       options.ShopID,
				Items:        cartResponse.CartItems,
				Delivery:     deliveryPayment[cartID].Delivery,
				Notes:        "заметочка",
				CoinIdsToUse: coinIdsToUse,
			})
	}

	if ammo.UsePromo && ammo.Promocode == nil {
		panic(fmt.Errorf("missing promocode parameter"))
	}

	req := &clients.CheckoutRequest{
		BuyerRegionID: options.RegionID,
		BuyerCurrency: "RUR",
		PaymentType:   "POSTPAID",
		PaymentMethod: deliveryPayment[0].PaymentMethod,
		Orders:        orderRequests,
		Buyer: clients.Buyer{
			LastName:   "последнееимя",
			FirstName:  "первоеимя",
			MiddleName: "среднееимя",
			Phone:      "+77777777777",
			Email:      options.Email,
			IP:         "8.8.8.8",
			DontCall:   true,
		},
		Promocode: ammo.Promocode,
	}

	resp, err := client.GetCheckouter().Checkout(shootContext, req, options, ammo.UID, ammo.UseFlash)
	if err != nil {
		panic(fmt.Errorf("cannot execute checkout: '%w'", err))
	}

	return resp
}

func GetAllowedCoins(cartRootResponse *clients.CartRootResponse, coinsForCart *clients.СoinsForCart) (*[]int, string) {
	var logMsg string
	coinIdsToUse := cartRootResponse.CoinIdsToUse
	var allowedCoinIds []int
	if coinIdsToUse == nil || len(*coinIdsToUse) == 0 {
		logMsg = "CoinIdsToUse null or Empty"
	} else if coinsForCart != nil {
		var applicableCoinIds []int
		for _, appCoin := range coinsForCart.ApplicableCoins {
			applicableCoinIds = append(applicableCoinIds, appCoin.ID)
		}
		for _, coinID := range *coinIdsToUse {
			isApplicable, _ := slices.Contains(applicableCoinIds, coinID)
			if isApplicable {
				allowedCoinIds = append(allowedCoinIds, coinID)
				continue
			} else {
				var reason string
				for restrictionType, disCoins := range coinsForCart.DisabledCoins {
					var disCoinIds []int
					for _, disCoin := range disCoins {
						disCoinIds = append(disCoinIds, disCoin.ID)
					}
					isDisabled, _ := slices.Contains(disCoinIds, coinID)
					if isDisabled {
						reason = restrictionType
						break
					}
				}
				logMsg = fmt.Sprintf("Can not use coin with id: %d, reason: %s", coinID, reason)
			}
		}
	}
	coinIdsToUse = &allowedCoinIds
	return coinIdsToUse, logMsg
}

func Unfreeze(client clients.ShootingClients, shootContext abstractions.ShootContext, options configs.Options, orderID int) error {
	if options.ForceUnfreeze {
		err := client.GetStockStorage().UnfreezeStocks(shootContext, orderID)
		if err != nil {
			return fmt.Errorf("cannot execute unfreeze: '%v'", err)
		}
	} else {
		err := client.GetCheckouter().Unfreeze(shootContext, orderID)
		if err != nil {
			return fmt.Errorf("cannot execute unfreeze: '%v'", err)
		}
	}
	return nil
}

func RecentOrders(client clients.CheckouterClient, shootContext abstractions.ShootContext, path string, uid int) {
	err := client.RecentOrders(shootContext, path, uid)
	if err != nil {
		panic(fmt.Errorf("cannot get recent orders: '%v'", err))
	}
}

func OptionsAvailabilitiesOfOrders(client clients.CheckouterClient, shootContext abstractions.ShootContext, path string, orderID int, uid int) {
	err := client.OptionsAvailabilitiesOfOrders(shootContext, orderID, path, uid)
	if err != nil {
		panic(fmt.Errorf("cannot get options-availabilities of orders: '%v'", err))
	}
}

func OrdersByUID(client clients.CheckouterClient, shootContext abstractions.ShootContext, path string, uid int) {
	err := client.OrdersByUID(shootContext, path, uid)
	if err != nil {
		panic(fmt.Errorf("cannot get orders by uid: '%v'", err))
	}
}

func GetOrdersByID1(client clients.CheckouterClient, shootContext abstractions.ShootContext, path string, orderID int, uid int) {
	query := &url.Values{}
	query.Add("clientRole", "SYSTEM")
	query.Add("clientId", strconv.Itoa(uid))
	query.Add("partials", "CHANGE_REQUEST")
	query.Add("partials", "CASHBACK_EMIT_INFO")

	getOrdersByID(client, model.LabelCheckouterOrdersByID, shootContext, path, orderID, query)
}

func GetOrdersByID2(client clients.CheckouterClient, shootContext abstractions.ShootContext, path string, orderID int) {
	query := &url.Values{}
	query.Add("clientRole", "SYSTEM")

	getOrdersByID(client, model.LabelCheckouterOrdersByID2, shootContext, path, orderID, query)
}

func getOrdersByID(client clients.CheckouterClient, label string, shootContext abstractions.ShootContext, path string, orderID int, query *url.Values) {
	err := client.GetOrdersByID(label, shootContext, path, query, orderID)
	if err != nil {
		panic(fmt.Errorf("cannot get orders by uid: '%v'", err))
	}
}

func GetOrders(client clients.CheckouterClient, shootContext abstractions.ShootContext, path string, uid int) {

	req := &clients.GetOrdersRequest{
		FromDate: "1970-00-00 00:00:00",
		PageInfo: clients.PageInfo{
			From:        1,
			To:          50,
			PageSize:    50,
			CurrentPage: 1,
		},
		Partials: []string{"CHANGE_REQUEST", "CASHBACK_EMIT_INFO"},
		RGBs:     []string{"BLUE", "WHITE"},
	}

	err := client.GetOrders(shootContext, req, path, uid)
	if err != nil {
		panic(fmt.Errorf("cannot execute getOrders: '%v'", err))
	}

}

func GetOfferFromIndex(client clients.ShootingClients, shootContext abstractions.ShootContext, options configs.Options, stock clients.Stock) (*clients.OfferInfo, error) {
	info, err := client.GetOTrace().GetOfferInfo(shootContext, stock, options)
	if err != nil {
		return nil, err
	}
	infoURL := info.Urls.Report.OfferInfoURL
	parsedURL, err := url.Parse(infoURL)
	if err != nil {
		return nil, err
	}
	offerInfo, err := client.GetReport().GetOfferInfoByURL(shootContext, parsedURL)

	if err != nil {
		return nil, err
	}
	return ConvertStockOfferFromReport(options, offerInfo.ReportSearch, stock), nil
}
