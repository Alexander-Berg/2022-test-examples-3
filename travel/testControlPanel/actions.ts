import {createAction} from 'typesafe-actions';

import {IApiRequestInfo} from 'server/utilities/TestRequestManager/types/requestInfo';

export const setApiRequestInfoItems = createAction(
    'test/setApiRequestInfoItems',
)<IApiRequestInfo[]>();
