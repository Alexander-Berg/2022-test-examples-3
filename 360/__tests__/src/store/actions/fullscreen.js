import { enterFullscreen, exitFullscreen } from 'store/actions/fullscreen';

it('enterFullscreen', () => {
    expect(enterFullscreen()).toMatchSnapshot();
});

it('exitFullscreen', () => {
    expect(exitFullscreen()).toMatchSnapshot();
});
