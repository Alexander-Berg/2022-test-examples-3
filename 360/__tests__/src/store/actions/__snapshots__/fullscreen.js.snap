// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enterFullscreen 1`] = `[Function]`;

exports[`exitFullscreen 1`] = `[Function]`;
